check/setup.go

```diff
@@ -353,30 +352,28 @@ func (check *HealthCheck) closeConnection(deleteTopicIfPresent bool) error {
 
 func (check *HealthCheck) deleteTopic(zkConn ZkConnection, chroot, name string, partitionID int32) error {
 	topic := ZkTopic{Name: name}
-	err := zkPartitions(&topic, zkConn, name, chroot)
+	err := zkPartitions(&topic, zkConn, topic.Name, chroot)
 	if err != nil {
 		return err
 	}
 
 	replicas, ok := topic.Partitions[partitionID]
 	if !ok {
-		return fmt.Errorf(`Cannot find partition with ID %d in topic "%s"`, partitionID, name)
+		return fmt.Errorf(`Cannot find partition with ID %d in topic "%s"`, partitionID, topic.Name)
 	}
 
 	brokerID := int32(check.config.brokerID)
 	if len(replicas) > 1 {
 		log.Info("Shrinking replication check topic to exclude broker ", brokerID)
 		replicas = delAll(replicas, brokerID)
-		return reassignPartition(zkConn, partitionID, replicas, name, chroot)
+		return reassignPartition(zkConn, partitionID, replicas, topic.Name, chroot)
 	}
 
-	log.Infof("topic %s has only one replica, deleting it", name)
+	log.Infof("topic %s has only one replica, deleting it", topic.Name)
 
-	acceptableTimeout := 3 * time.Second
+	r := retrier.New(retrier.ConstantBackoff(3, 100*time.Millisecond), nil)
```

Comment:
`100ms` between each retry is a bit too few, I would do at least `1000ms` (1 sec) and maybe we can use `ExponentialBackoff` too?
like ExponentialBackoff(3, 1000 * time.Millisecond).

If we parameterise the method, testing it will be easier (and faster) too

Reply:
Also the you may use the  `retryInterval`  in the config object..

Reply:
Maybe not. THe `retryInterval` is 1ms only. But I am in favor of using the config to store this values.